### PR TITLE
feat: #1031 Enable SchemaSpy to auto generate data model

### PR DESCRIPTION
**Pull request by MCatherine1994**
refs: #1031

### CODE REVIEW

File: .github/workflows/reusable_data_model_gen.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    steps:
+      - uses: actions/checkout@v4
+      - name: Flyway
+        uses: docker://flyway/flyway:10
+        env:
+          FLYWAY_URL: jdbc:postgresql://postgres:5432/fam
+          FLYWAY_USER: ${{ needs.vars.outputs.user }}
+          FLYWAY_PASSWORD: ${{ needs.vars.outputs.pass }}
+          FLYWAY_LOCATIONS: filesystem:./server/flyway/sql
+          FLYWAY_DEFAULT_SCHEMA: "app_fam"
```
Comment (MCatherine1994) [inline .github/workflows/reusable_data_model_gen.yml:52]: Database name is "fam" to match our docker (shouldn't matter though, as long as we used the same name below when run the flyway. In terraform we deployed our db named "famdb"). 

Database schema is "app_fam"

username and password as long as match the ones we used when create the database will be good

File: .github/workflows/reusable_data_model_gen.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
+name: Reusable Data Model Generator
+
+on:
+  workflow_dispatch:
```
Comment (MCatherine1994) [inline .github/workflows/reusable_data_model_gen.yml:4]: For now, make this pipeline to be triggered manually, want to test how it works first

File: .github/workflows/reusable_data_model_gen.yml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+          echo "user=$USERNAME" >> $GITHUB_OUTPUT
+          echo "pass=$PASSWORD" >> $GITHUB_OUTPUT
+
+  schemaspy:
+    name: Generate Documentation
+    runs-on: ubuntu-22.04
+    needs: [vars]
+    services:
+      postgres:
+        image: postgis/postgis:16-3.4
```
Comment (ianliuwk1019) [inline .github/workflows/reusable_data_model_gen.yml:30]: Wondering if we just need basic postgres image than postgis image for FAM?

Comment (MCatherine1994) [reply .github/workflows/reusable_data_model_gen.yml]: yeah I'll change to the same one from our docker, thanks!

**Review by ianliuwk1019 — APPROVED** [review]

**Pull request merged**